File: src/JavaScript/sample1.js
// Sample JavaScript file with intentional code quality issues

function processData(data) {
    // Unused variable
    var unusedVar = "This is never used";
    
    // Using var instead of const/let
    var result = data + " processed";
    
    // Missing semicolons
    console.log(result)
    
    return result
}

// Function complexity issue
function complexFunction(a, b, c, d) {
    if (a > 0) {
        if (b > 0) {
            if (c > 0) {
                if (d > 0) {
                    return a + b + c + d;
                } else {
                    return a + b + c - d;
                }
            }
        }
    }
    return 0;
}

// Magic numbers
function calculate() {
    var result = 42 * 100 + 3.14159;
    return result;
}

// No error handling
function riskyOperation() {
    var data = JSON.parse("invalid json");
    return data;
}

module.exports = {
    processData,
    complexFunction,
    calculate
};

File: src/Python/sample1.py
"""
Sample Python file with intentional code quality issues
"""

# Unused import
import sys
import os

# Global variable - bad practice
GLOBAL_VAR = "global"

def function_with_issues(data):
    """Process data with issues."""
    # Unused variable
    unused_variable = "never used"
    
    # Bare except - bad practice
    try:
        result = int(data)
    except:
        pass
    
    # Magic number
    return data * 42

def complex_function(a, b, c, d):
    """Function with high complexity."""
    if a > 0:
        if b > 0:
            if c > 0:
                if d > 0:
                    return a + b + c + d
                else:
                    return a + b + c - d
            else:
                return a + b - c + d
        else:
            return a - b + c + d
    else:
        return -a + b + c + d

# No docstring
def no_documentation():
    print("This function has no documentation")

# Line too long
def function_with_long_line():
    very_long_variable_name = "This is a very long string that exceeds the recommended line length and should be split into multiple lines for better readability"
    return very_long_variable_name

if __name__ == "__main__":
    function_with_issues("test")
